languages/csharp/exercises/concept/inheritance/Inheritance.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+
+abstract class Character
+{
+    protected Character(string characterType)
+    {
+        throw new NotImplementedException("Please implement the Character() constructor");
+    }
+
+    public abstract int DamagePoints(Character target);
+
+    public virtual bool Vulnerable()
+    {
+        throw new NotImplementedException("Please implement the Character.Vulnerable() method");
+    }
+
+    public override string ToString()
+    {
+        throw new NotImplementedException("Please implement the Character.ToString() method");
+    }
+}
+
+class Warrior : Character
+{
+    public Warrior() : base("TODO")
+    {
+    }
+
+    public override int DamagePoints(Character target)
+    {
+        throw new NotImplementedException("Please implement the Warrior.DamagePoints() method");
+    }
+}
+
+class Wizard : Character
```

Comment:
I did not provide a stub for the overriding of the `Vulnerable` method, as I'd like students to figure that out themselves. Thoughts on that?

Reply:
Agree.

Reply:
You could provide no stubs in the `Wizard` at all...

Reply:
But then the code would not compile, as `DamagePoints` is abstract.

Reply:
Dammed compiled languages ;)

I was just thinking that the Warrior class might act as a nice sort of template for the Wizard one.

Reply:
Yeah, I went back and forth on this, but I think having the stubs in place makes sense.